src/pages/ExternalSupplierInvoicePage.js

```diff
@@ -251,53 +263,14 @@ export class ExternalSupplierInvoicePage extends GenericPage {
   renderCell(key, transactionBatch) {
     const isEditable = !this.props.transaction.isFinalised;
     const type = isEditable ? 'editable' : 'text';
-    switch (key) {
-      default:
-        return transactionBatch[key];
-      case 'packSize': {
-        const renderedCell = {
-          type: type,
-          cellContents: String(transactionBatch.packSize),
-        };
-        return renderedCell;
-      }
-      case 'numberOfPacks': {
-        const renderedCell = {
-          type: type,
-          cellContents: String(transactionBatch.numberOfPacks),
-        };
-        return renderedCell;
-      }
-      case 'batch': {
-        const renderedCell = {
-          type: type,
-          cellContents: transactionBatch.batch,
-          keyboardType: 'default',
-        };
-        return renderedCell;
-      }
-      case 'costPrice': {
-        const renderedCell = {
-          type: type,
-          cellContents: String(transactionBatch.costPrice),
-        };
-        return renderedCell;
-      }
-      case 'expiryDate': {
-        const expiryDate = formatExpiryDate(transactionBatch.expiryDate);
-        const renderedCell = {
-          type: type,
-          cellContents: expiryDate || 'month/year',
-        };
-        return renderedCell;
-      }
-      case 'remove':
-        return {
-          type: 'checkable',
-          icon: 'md-remove-circle',
-          isDisabled: this.props.transaction.isFinalised,
-        };
-    }
+
+    const renderCellGenerator =
+      renderCellMapping[key] || defaultRenderCellWrapper;
```

Comment:
what do you think about this kind of mapping ? 
Ideally defaults would go somewhere common.
And remove (in next commit) can go into renderCellMapping, and instead of being disabled, can change type to text or ideally .. hide the column all together on finalised